code/controllers/subsystem/ships.dm

```diff
@@ -119,7 +119,7 @@ var/global/list/ftl_weapons_consoles = list()
 		for(var/datum/component/weapon/W in S.components)
 			if(world.time > W.next_attack && W.fire_rate)
 				W.next_attack = world.time + W.fire_rate + rand(1,100)
-				attack_player(S, W.attack_data)
+				attack_player(S, W.attack_data, W)
```

Comment:
replace this with attack_player(S, W) and make a new attack data var inside wherever it's used